pandas/core/resample.py

```diff
@@ -964,6 +965,7 @@ def __init__(self, obj, *args, **kwargs):
         self._groupby.grouper.mutated = True
         self.groupby = copy.copy(parent.groupby)
 
+    @no_type_check
```

Comment:
This file has a set of mismatched signatures that's much more involved to bring into alignment that the ones we did in other PRs.

# `_upsample`
`DatetimeIndexResampler` and `Resampler` both define an `_upsample` method. The two signatures are only marginally different and could be harmonized easily. But `_GroupByMixin` defines an `_apply` method that is significantly different and then assigns that method to the class's `_upsample` variable (as well as `_downsample` and `_groupby_and_aggregate`).

# `_downsample`
`_downsample has similar relationships between `PeriodIndexResampler`, `DatetimeIndexResampler`, and `_GroupByMixin`.

# `_groupby_and_aggregate`
`_groupby_and_aggregate` has similar relationships between `Resampler` and `_GroupByMixin`.

Some parts of the `_apply` signatures appear to be unused in this file. I'm investigating how much mucking around I can do with them. It might be possible to bring them all in alignment.

Reply:
@gwrome there is lots of magic to make grouby/resample/rolling play nicely. But can of course do this another time.